lib/context_adapter_server.js

```diff
@@ -274,6 +319,11 @@ function requestHandler(request, reply) {
 
   if (caComm.isPollingUpdateContext(request)) {
     respond2Request(request, reply);
+  } else if (caComm.getCellIdDescriptors(request).length) {
+    async.parallel([
+      respond2Request.bind(null, request, reply),
+      processCellIdRequest.bind(null, request)
```

Comment:
Both are independent operations, the respond2Request() is just to respond to the request received by the CB and the processCellIdRequest() is the one which gets the geolocation and sends a new updateContext request to the CB... So, they are 100% independent operations... Anyhow, no problem on my side to execute them in sequence... :) Any preference or shall I leave as it is?